rust/projects/parallelism/tests/thread_pool.rs

```diff
@@ -0,0 +1,64 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
+use kvs::thread_pool::*;
+use kvs::Result;
+
+use crossbeam_utils::sync::WaitGroup;
```

Comment:
Nice to use this since it casually introduces the user to another synchronization primitive.